### PR TITLE
Adding Terraform related information to Radius FAQ page

### PR DESCRIPTION
**Pull request by jasonviviano**
Thank you for helping make the Radius documentation better!

**Please follow this checklist before submitting:**

- [ ] [Read the contribution guide](https://docs.radapp.dev/contributing/docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

- Added Terraform related info to: 
1) What languages does Radius support?
2) What resources can Recipes deploy?
3) Does Radius support GCP?

### Summary
📝📊🔧

This pull request updates and improves the docs project, adding a new section on deploying applications, removing outdated content on Dapr, aligning the quickstarts and tutorials with the latest CLI usage, fixing spelling and formatting issues, and applying the Diátaxis framework terminology.

> _The docs team has been very busy_
> _Updating the content for Radius_
> _They added deploy apps_
> _And fixed some code snags_
> _And made the docs more user-friendly_

### Walkthrough
*  Add a new section on deploying applications with overview, how-to, and reference pages ([link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-4ec4dc25157505e6517a7256e5c1e20697107e4b31be583dc1107c71cc27ff1aL1-R6), [link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-41c32a6cceb1df53325787dba357f4b12e474d70425200b0e0e0579a27b74e07L1-R45), [link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-63b37f4df2fc15989682bb471b894aac2f0f3e248e60968a684f92ff6826ae65R1-R41))
*  Move the `initialize-environment.md` file from `author-apps/dev-environment` to `deploy-apps/dev-environment` and update the `weight` and the `rad init` command ([link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-0737bce6e737e110e224fe4ed79ed0a4c93df2f7acc6c6220840f38aa23eca4bL6-R6), [link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-0737bce6e737e110e224fe4ed79ed0a4c93df2f7acc6c6220840f38aa23eca4bL28-R30), [link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-0737bce6e737e110e224fe4ed79ed0a4c93df2f7acc6c6220840f38aa23eca4bL86))
*  Update the `baseURL`, `version`, `tag_version`, `chart_version`, and `github_branch` in the docs configuration file to use `edge` instead of `v0.23` ([link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-5fbd9920d4a0ae5420b9709cdc967f57514008cae257ca8735d7b8c02e1c77f2L1-R1), [link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-5fbd9920d4a0ae5420b9709cdc967f57514008cae257ca8735d7b8c02e1c77f2L63-R65), [link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-5fbd9920d4a0ae5420b9709cdc967f57514008cae257ca8735d7b8c02e1c77f2L73-R73))
*  Update the title and linkTitle of the `author-apps/_index.md` file to use `Authoring applications` instead of `Developing applications` ([link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-20dca618be0475186b3ec55a7aa672dd41822c1d178a066e20eb43d5a321e500L3-R4))
*  Add a new paragraph and an image to the `application/index.md` file explaining how Radius automates the configuration of connections ([link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-66965bf31451ac9af6355a1a15f187c30956fd16b35bfc34e665556b34a3d99bR15-R20))
*  Update the `container/index.md` file to explain how connections inject environment variables into containers and how to override them ([link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-9b2dadf764d06565222c44220d12e2c65fed52e670dde1c3d2d2f433c20dc833L73-R89))
*  Update the `weight` of the `community.md` and `contributing/_index.md` files to adjust the order of the menu items in the docs sidebar ([link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-3c48e1220fbb9bf70f57f430bc2d1f7992e8ce3b55890da51bafb0eeb466538fL6-R6), [link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-8f6693da53062b2a0040aae060f7f2020b178bb5b5f888c679e803ed68109f1eL6-R6))
*  Update the `application-graph/index.md` file to remove the redundant paragraph about automation and provide a link to the deployment section ([link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-0fc8a723eda4e6ccfee5a91b1b44ba950a2792fa048faa220e64b4f671c22730L17-R20))
*  Rename the `contributing-docs.md` file to `contributing-docs/index.md` and add a new paragraph and an image explaining the Diátaxis framework, simplify the steps for generating the CLI reference docs, fix the numbering of the steps for setting up the docs environment, and reorganize the sections on types of docs, style and tone, spelling, and tips and tricks ([link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-cf6a91b553874620b91d843c22c9a758834b55df96f7e5ffbcdb17fb855419cdL12-R23), [link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-cf6a91b553874620b91d843c22c9a758834b55df96f7e5ffbcdb17fb855419cdL49-R67), [link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-cf6a91b553874620b91d843c22c9a758834b55df96f7e5ffbcdb17fb855419cdL71-R73), [link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-cf6a91b553874620b91d843c22c9a758834b55df96f7e5ffbcdb17fb855419cdL90-R162))
*  Update the `first-app/_index.md` and the quickstart files to use the `rad init` and `rad init --full` commands according to the latest changes in the CLI ([link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-53b840e80ae25169294077f38fc40ea37ef23fced8cd3345ad07bceaa6b812e7L72-R72), [link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-53b840e80ae25169294077f38fc40ea37ef23fced8cd3345ad07bceaa6b812e7L129-R129), [link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-411f6f244e028f30542e3a3a81cb20d8d50bd60b1587656dd7afcd6a014f12b2L39-R42), [link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-707f3b4e819a3b98ec9a1b0745e8f6a78e32144b7ab7973fc94137370da7733eL28-R31))
*  Delete the files related to the quickstart-dapr section, which was removed from the docs ([link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-0a5c54beb9b7863c868051acb8f589e6f806ef9fe62af8756cccf6ea94a8923b), [link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-ef2c4ae552c5bec5e6d0bd4b5761a1516d7de9597094e5e4775f880e6c4f6611), [link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-82ee426db5ea4a071df4465003c8fc8c3c49188baa0826d1c9c807b6638dda10), [link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-79a518d1fc53fc5e56e65c936aad3b842f57ea6d3b8564f4b5041e053b5857a9))
*  Add several words to the custom spelling dictionary to avoid spelling errors in the docs ([link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-f46255e63294aaad4ef89f0874b1a4b76bd218f71b938b7af3d36c02290dad84R253), [link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-f46255e63294aaad4ef89f0874b1a4b76bd218f71b938b7af3d36c02290dad84R429-R432), [link](https://github.com/project-radius/docs/pull/691/files?diff=unified&w=0#diff-f46255e63294aaad4ef89f0874b1a4b76bd218f71b938b7af3d36c02290dad84L435-R451))

### CODE REVIEW

File: docs/content/reference/faq.md [hunk 1]
```diff
@@ -22,9 +22,23 @@ Open-source Radius requires that you self-host and run your own Radius instance
 
 ### What languages does Radius support? 
 
-For core application source code, Radius supports any language as long as it is containerized. 
+#### Core application source code
```
Comment (willtsai) [inline docs/content/reference/faq.md:25]: Recommend getting rid of these subheaders and use bullet points combined with bolding instead.

File: docs/content/reference/faq.md [hunk 2]
```diff
@@ -22,9 +22,23 @@ Open-source Radius requires that you self-host and run your own Radius instance
...
 
-For Radius application and environment definitions, Radius currently only supports Bicep. Terraform is on our backlog and is being investigated. Stay tuned for updates. 
+Radius supports any language as long as it is containerized. 
+
+#### Radius Application and Radius Environment definitions
+
+| Infrastructure as Code Languages | Radius Support |
```
Comment (willtsai) [inline docs/content/reference/faq.md:31]: I like these tables, but would recommend changing the subheaders into bullet points

File: docs/content/reference/faq.md [hunk 3]
```diff
@@ -43,7 +57,7 @@ Users can employ multiple environments for isolation and organization, for examp
 
 ### What resources can Recipes deploy? 
 
-Recipes support any resource that can be modeled in Bicep. Terraform support is on our backlog and is being investigated. Stay tuned for updates.
+Radius Recipes support any resources that can be modeled in Bicep or that has an applicable Terraform provider. Radius can use any Terraform provider.
```
Comment (AaronCrawfis) [inline docs/content/reference/faq.md:60]: We should validate that we can use any provider, such as GCP. Passing in the provider configuration needs to be tested.

**Comment by jasonviviano** [comment]
Closing in favor of: https://github.com/project-radius/docs/pull/700/files

**Pull request closed without merging**